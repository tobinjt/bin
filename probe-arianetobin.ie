#!/bin/bash

set -e -f -u -o pipefail

NUMBER_OF_ATTEMPTS=5
SECONDS_TO_SLEEP_BETWEEN_ATTEMPTS=600

main() {
  SENTINEL='<title>Ariane Tobin Jewellery - Ariane Tobin Jewellery</title>'
  WEBSITE='https://www.arianetobin.ie/'
  # Try multiple times to cover for bad internet at home.
  for _ in $(seq 1 "${NUMBER_OF_ATTEMPTS}"); do
    # Redirect to /dev/null rather than using -q: using -q means grep will exit
    # immediately if a match is found, and then curl can't finish writing to the
    # pipe so it produces an error message, whereas without -q grep will process
    # all the input.
    if curl --silent "${WEBSITE}" | grep -F -i "${SENTINEL}" > /dev/null; then
      return 0
    fi
    sleep "${SECONDS_TO_SLEEP_BETWEEN_ATTEMPTS}"
  done
  printf "Did not find sentinel '%s' in %s\\n" "${SENTINEL}" "${WEBSITE}"
  return 1
}

# Only run main if being executed directly; do nothing if sourced for testing.
# Use basename so that it can be run as 'bash -x ./program ...'.
if [[ "$(basename "${BASH_SOURCE[0]}")" == "$(basename "${0}")" ]]; then
  main
fi
