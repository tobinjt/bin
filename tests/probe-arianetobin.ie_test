#!/bin/bash

set -e -f -u -o pipefail

source ../probe-arianetobin.ie

SECONDS_TO_SLEEP_BETWEEN_ATTEMPTS=0
export SECONDS_TO_SLEEP_BETWEEN_ATTEMPTS

stdout="$(mktemp -t stdout.XXXXXXXXXXXXXXXXX)"
stderr="$(mktemp -t stderr.XXXXXXXXXXXXXXXXX)"
{
  # Success on first attempt.
  curl() {
    printf "%s\\n" "${SENTINEL}"
  }
  main || print "Unexpected failure of test 1\\n"

  # Repeated failures.
  curl() {
    printf "No sentinel here\\n"
    printf "curl was invoked for test 2\\n" >&2
  }
  main || printf "Test 2: main failed as expected\\n"
  # TODO: write a test where the first attempt fails and the second succeeds,
  # like in check-dns-for-hosting_test.  I've tried to do this without success,
  # because curl() is not redefined like host is, and I don't understand why :(
} > "${stdout}" 2> "${stderr}" || true

diff -u "probe-arianetobin.ie_test.stdout" "${stdout}"
diff -u "probe-arianetobin.ie_test.stderr" "${stderr}"
rm -f "${stdout}" "${stderr}"
touch .probe-arianetobin.ie_timestamp
