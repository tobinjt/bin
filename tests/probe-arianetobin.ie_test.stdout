Test 2: main failed as expected
